packages/portal-proto/src/features/GenomicTables/SomaticMutationsTable/SMTableContainer.tsx

```diff
@@ -485,11 +536,19 @@ export const SMTableContainer: React.FC<SMTableContainerProps> = ({
                 />
 
                 {caseFilter || geneSymbol ? (
-                  <ButtonTooltip label="Export current view" comingSoon={true}>
-                    <FunctionButton data-testid="button-tsv-mutation-frequency">
-                      TSV
-                    </FunctionButton>
-                  </ButtonTooltip>
+                  <FunctionButton
+                    data-testid="button-tsv-mutation-frequency"
+                    onClick={
+                      caseFilter ? handleTSVCaseDownload : handleTSVGeneDownload
+                    }
+                  >
+                    {downloadMutationsFrequencyTSVCaseActive ||
```

Comment:
Seems like you only need one piece of state here (`downloadMutationsFrequencyTSVActive` maybe?) instead of two.

Reply:
You are correct. I removed it. :)